pkg/controllers/provisioning/scheduling/scheduling_benchmark_test.go

```diff
@@ -259,15 +285,16 @@ func makePodAntiAffinityPods(count int, key string) []*corev1.Pod {
 func makePodAffinityPods(count int, key string) []*corev1.Pod {
 	var pods []*corev1.Pod
 	for i := 0; i < count; i++ {
+		labels := randomAffinityLabels() // self-affinity
```

Comment:
At thousands of pods scale, were we having collisions?

Reply:
It's not that we were having collisions, it's that at smaller scale, you don't guarantee that there is a pod in every domain -- in order for the affinity to be satisfied, you have to have a pod in that domain (unless you are using self-affinity, then you are allowed to schedule the pod so long as there are already no pods across all domains)